### PR TITLE
Adding spatial index 0.2.1

### PR DESCRIPTION
**Pull request by ferdonline**
Deploying Spatial Index 0.2.1 after Weina's additions.
 - SPIND-82: Query window for Spatial Index
 - SPIND-84: Spatial Index Point API
- Add `has_Soma` flag (default=true) in `add_neuron` to allow the API to add segments only (compat with data from MorphIO)
 - Add `add_soma` function in MorphIndex API
 - Add `__str__` in the Python API

### CODE REVIEW

File: var/spack/repos/builtin/packages/spatial-index/package.py [hunk 1]
```diff
@@ -16,7 +15,9 @@ class SpatialIndex(PythonPackage):
     depends_on("cmake")
     depends_on("boost")
 
-    version('0.2.0', tag='0.2.0', submodules=True)
+    version('0.2.1', tag='0.2.1', submodules=True)
     version('0.1.0', tag='0.1.0', submodules=True)
 
-
+    @run_after('install')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/spatial-index/package.py:21]: Hmm, I guess it's either that or we make it a CMakePackage and jump through additional hoops to get the Python setup right :/

**Review by matz-e — APPROVED** [review]

File: var/spack/repos/builtin/packages/spatial-index/package.py (hunk 1 above)
Comment (pramodk) [reply var/spack/repos/builtin/packages/spatial-index/package.py:21]: yeah...this is "non-standard" (using post-install hook to "complete" part of installation)

Comment (ferdonline) [reply var/spack/repos/builtin/packages/spatial-index/package.py:21]: We could override install(), call `super().install()` and the `install_tree`... But IMO it's complicating...

Comment (matz-e) [reply var/spack/repos/builtin/packages/spatial-index/package.py:21]: It's fine as it is, a nice solution :)

I was just lamenting the extra complexity necessary with either Python and CMake packages to get hybrids dealt with.

**Pull request merged**